Field policy merge docs (#9969)
apollographql/apollo-client: docs/source/caching/cache-field-behavior.mdx

Remember that mergeable objects will only be merged with existing objects occupying the same field of the same parent object, and only when the `__typename` of the objects is the same. If you really need to work around these rules, you can write a custom `merge` function to do whatever you want, but `merge: true` follows these rules.

> In some cases you may want to completely disable merge functions for certain fields and you can do so by setting a merge function to `merge: false`.

Reviewer: ```suggestion
In some cases, you might want to completely disable merge functions for certain fields. To do so, pass `merge: false` like so:
```
